homeassistant/helpers/script.py

```diff
@@ -539,6 +549,64 @@ async def _async_choose_step(self):
         if choose_data["default"]:
             await self._async_run_script(choose_data["default"])
 
+    async def _async_wait_for_trigger_step(self):
+        """Wait for a trigger event."""
+        if CONF_TIMEOUT in self._action:
+            delay = self._get_pos_time_period_template(CONF_TIMEOUT).total_seconds()
+        else:
+            delay = None
+
+        self._script.last_action = self._action.get(CONF_ALIAS, "wait for trigger")
+        self._log(
+            "Executing step %s%s",
+            self._script.last_action,
+            "" if delay is None else f" (timeout: {timedelta(seconds=delay)})",
+        )
+
+        variables = deepcopy(self._variables)
+        self._variables["wait"] = {"remaining": delay, "trigger": None}
+
+        async def async_done(variables, context=None):
+            self._variables["wait"] = {
```

Comment:
Should this be `wait_for_trigger` to be aligned with the key that triggers the script ?

Reply:
The idea is there's a `wait` variable that indicates the result of the last wait, no matter which type. So it applies to both `wait_template` and `wait_for_trigger`, whichever happened last. That's how it's documented (or will be soon, once I get to update the doc change via a rebase, etc.) To be clear, there's already a variable named `wait` that so far applies to `wait_template`.